requestfinder.py

```diff
@@ -0,0 +1,226 @@
+                            if review.by_project.startswith(p):
+                                skip = True
+                                break
+                    if not skip:
+                        if review.by_package:
+                            print "osc review %s -m '%s' -P %s -p %s %s" % (settings['action'], settings['message'], review.by_project, review.by_package, r.reqid)
+                        else:
+                            print "osc review %s -m '%s' -P %s %s" % (settings['action'], settings['message'], review.by_project, r.reqid)
+                elif review.by_group:
+                    print "osc review %s -m '%s' -G %s %s" % (settings['action'], settings['message'], review.by_group, r.reqid)
+                elif review.by_user:
+                    skip = False
+                    if settings['exclude-user']:
+                        for u in settings['exclude-user']:
+                            if review.by_user == u:
+                                skip = True
+                                break
+                    if not skip:
+                        print "osc review %s -m '%s' -U %s %s" % (settings['action'], settings['message'], review.by_user, r.reqid)
+
+    @cmdln.option('--query', metavar='filterstr', help='filter string')
+    @cmdln.option('--action', metavar='action', help='action (accept/decline)')
+    @cmdln.option('--settings', metavar='settings', help='settings to load from config file')
+    @cmdln.option('-m', '--message', metavar="message", help="message")
+    def do_request(self, subcmd, opts):
+        """${cmd_name}: print commands for requests
+
+        ${cmd_usage}
+        ${cmd_option_list}
+        """
+
+        settings = {
+            'action': 'reopen',
+            'message': 'reopen',
+            'query': None,
+        }
+
+        if opts.settings:
+            self._load_settings(settings, opts.settings)
+
+        rqs = self.tool.find_requests(settings['query'])
+        for r in rqs:
+            print '#', r.reqid, r.get_creator(), r.actions[0].src_project, r.actions[0].src_package, r.actions[0].tgt_project
+            print "osc rq {} -m '{}' {}".format(settings['action'], settings['message'], r.reqid)
+
+    def help_examples(self):
+        return """$ cat > ~/.config/opensuse-release-tools/requestfinder.conf << EOF
+        [settings foo]
+        query = (review[@by_project='example' and @state='new']
+                 and state/@name='review'
+                 and action/source/@project='openSUSE:Factory'
+                 and action/target/@project='openSUSE:Leap:15.0'
+        exclude-user = repo-checker
+        exclude-project = openSUSE:Leap:15.0:Staging
+        message = override
+        action = accept
+        EOF
+        $ ${name} review --settings foo | tee doit.sh
+        ./doit.sh
+        """
+
+if __name__ == "__main__":
+    app = CommandLineInterface()
+    sys.exit(app.main())
+
+# vim: sw=4 et
```

Comment:
Is there no project level configuration file that can be added for this rather than including in all files? Still never get this as we could add one for every editor folks use.

Reply:
there are emacs tags too so this is just fair. There are only those two editors, the rest is toys anyways :p

Reply:
Religious debates aside is there no way to configure that for an entire directory? Like `.vimrc` or somesuch?

Reply:
with some extra code in .vimrc yes

Reply:
So we have something like 30 lines of boilerplate per file before we even get to python? Eww.